conda_lock/models/lock_spec.py

```diff
@@ -61,6 +62,7 @@ class LockSpecification(BaseModel):
     # TODO: Should we store the auth info in here?
     channels: List[Channel]
     sources: List[pathlib.Path]
+    pip_repositories: List[PipRepository] = Field(default_factory=list)
```

Comment:
It certainly does support this, however I generally avoid using it as it may lead readers of the code to believe that the language in general works this way. Happy to change it if you feel strongly.